Commit 1

leushkinml opened the pull request.

leushkinml committed:
Commit 1

leushkinml committed:
Commit 2    17.01.2023    18:51

leushkinml committed:
Commit 3    18.01.2023    23:35
Подключена база данных.

Commit 2    17.01.2023    18:51

leushkinml committed:
Commit 3    18.01.2023    23:35
Подключена база данных.

Commit 2    17.01.2023    18:51

kalyashov commented on src/main/java/ru/hogwarts/schoollion/controller/StudentController.java:
```diff
@@ -0,0 +1,84 @@
+package ru.hogwarts.schoollion.controller;
+
+
+import io.swagger.v3.oas.annotations.parameters.RequestBody;
```
Убери этот импорт, импорт для аннотации RequestBody уже есть ниже, а этой аннотацией ты подтягиваешь неверный RequestBody

Это относится к обоим контроллерам

kalyashov commented on src/main/java/ru/hogwarts/schoollion/controller/StudentController.java:
```diff
@@ -0,0 +1,84 @@
+        return ResponseEntity.ok(studentService.findStudentByAge(studentAge));
+    }
+
+//    @GetMapping("{studentAge}")   // GET http://localhost:8080/student/18
```
Убери закомментированный код, как вижу он уже не нужен

kalyashov replied:
Это относится ко всем классам

kalyashov commented on src/main/java/ru/hogwarts/schoollion/service/StudentService.java:
```diff
@@ -0,0 +1,88 @@
+//        return result;
+//    }
+    public List<Student> findStudentByAge(int age) {
+        return studentRepository.findStudentByAge(age);
```
А здесь стоит добавить проверки - если передан отрицательный возраст

kalyashov commented on src/main/resources/application.properties:
```diff
@@ -1 +1,6 @@
+spring.datasource.username= student
+spring.datasource.password= chocolatefrog
 
+Hibernate=ddl auto (create, create-drop, validate, update)
```
Это не относится к конфигурации, надо убрать

leushkinml committed:
Commit 4    19.01.2023    23:39

Commit 3    18.01.2023    23:35
Подключена база данных.

Commit 2    17.01.2023    18:51

leushkinml committed:
Commit 5    20.01.2023    08:39
Не работают
1. Получить факультет студента
2. Получить студентов факультета

Commit 4    19.01.2023    23:39
Устранил некорректный импорт

Commit 3    18.01.2023    23:35
Подключена база данных.

Commit 2    17.01.2023    18:51

leushkinml committed:
Commit 5    20.01.2023    08:39
Не работают
1. Получить факультет студента
2. Получить студентов факультета

Commit 4    19.01.2023    23:39
Устранил некорректный импорт

Commit 3    18.01.2023    23:35
Подключена база данных.

Commit 2    17.01.2023    18:51

leushkinml committed:
Commit 7    23.01.2023    20:07
ДЗ выполнено.

Commit 6    22.01.2023    20:28
ДЗ выполнено.

Commit 5    20.01.2023    08:39
Не работают
1. Получить факультет студента
2. Получить студентов факультета

Commit 4    19.01.2023    23:39
Устранил некорректный импорт

Commit 3    18.01.2023    23:35
Подключена база данных.

Commit 2    17.01.2023    18:51

leushkinml committed:
Commit 7    23.01.2023    20:07
ДЗ выполнено.

Commit 6    22.01.2023    20:28
ДЗ выполнено.

Commit 5    20.01.2023    08:39
Не работают
1. Получить факультет студента
2. Получить студентов факультета

Commit 4    19.01.2023    23:39
Устранил некорректный импорт

Commit 3    18.01.2023    23:35
Подключена база данных.

Commit 2    17.01.2023    18:51

leushkinml committed:
Commit 8    26.01.2023    18:10
Потоки данных. Работа с файлами.

Commit 7    23.01.2023    20:07
ДЗ выполнено.

Commit 6    22.01.2023    20:28
ДЗ выполнено.

Commit 5    20.01.2023    08:39
Не работают
1. Получить факультет студента
2. Получить студентов факультета

Commit 4    19.01.2023    23:39
Устранил некорректный импорт

Commit 3    18.01.2023    23:35
Подключена база данных.

Commit 2    17.01.2023    18:51

leushkinml committed:
Commit 8    26.01.2023    18:10
Потоки данных. Работа с файлами.

Commit 7    23.01.2023    20:07
ДЗ выполнено.

Commit 6    22.01.2023    20:28
ДЗ выполнено.

Commit 5    20.01.2023    08:39
Не работают
1. Получить факультет студента
2. Получить студентов факультета

Commit 4    19.01.2023    23:39
Устранил некорректный импорт

Commit 3    18.01.2023    23:35
Подключена база данных.

Commit 2    17.01.2023    18:51

leushkinml committed:
Commit 9    28.01.2023    22:43
Тесты контроллеров

Commit 8    26.01.2023    18:10
Потоки данных. Работа с файлами.

Commit 7    23.01.2023    20:07
ДЗ выполнено.

Commit 6    22.01.2023    20:28
ДЗ выполнено.

Commit 5    20.01.2023    08:39
Не работают
1. Получить факультет студента
2. Получить студентов факультета

Commit 4    19.01.2023    23:39
Устранил некорректный импорт

Commit 3    18.01.2023    23:35
Подключена база данных.

Commit 2    17.01.2023    18:51

leushkinml committed:
Commit 10    30.01.2023    20:39

Commit 9    28.01.2023    22:43
Тесты контроллеров

Commit 8    26.01.2023    18:10
Потоки данных. Работа с файлами.

Commit 7    23.01.2023    20:07
ДЗ выполнено.

Commit 6    22.01.2023    20:28
ДЗ выполнено.

Commit 5    20.01.2023    08:39
Не работают
1. Получить факультет студента
2. Получить студентов факультета

Commit 4    19.01.2023    23:39
Устранил некорректный импорт

Commit 3    18.01.2023    23:35
Подключена база данных.

Commit 2    17.01.2023    18:51

leushkinml committed:
Commit 10    30.01.2023    20:39

Commit 9    28.01.2023    22:43
Тесты контроллеров

Commit 8    26.01.2023    18:10
Потоки данных. Работа с файлами.

Commit 7    23.01.2023    20:07
ДЗ выполнено.

Commit 6    22.01.2023    20:28
ДЗ выполнено.

Commit 5    20.01.2023    08:39
Не работают
1. Получить факультет студента
2. Получить студентов факультета

Commit 4    19.01.2023    23:39
Устранил некорректный импорт

Commit 3    18.01.2023    23:35
Подключена база данных.

Commit 2    17.01.2023    18:51

leushkinml committed:
Commit 11    02.02.2023    15:12

Commit 10    30.01.2023    20:39

Commit 9    28.01.2023    22:43
Тесты контроллеров

Commit 8    26.01.2023    18:10
Потоки данных. Работа с файлами.

Commit 7    23.01.2023    20:07
ДЗ выполнено.

Commit 6    22.01.2023    20:28
ДЗ выполнено.

Commit 5    20.01.2023    08:39
Не работают
1. Получить факультет студента
2. Получить студентов факультета

Commit 4    19.01.2023    23:39
Устранил некорректный импорт

Commit 3    18.01.2023    23:35
Подключена база данных.

Commit 2    17.01.2023    18:51

leushkinml committed:
Commit 12    03.02.2023    14:20
SQL запросы

Commit 11    02.02.2023    15:12

Commit 10    30.01.2023    20:39

Commit 9    28.01.2023    22:43
Тесты контроллеров

Commit 8    26.01.2023    18:10
Потоки данных. Работа с файлами.

Commit 7    23.01.2023    20:07
ДЗ выполнено.

Commit 6    22.01.2023    20:28
ДЗ выполнено.

Commit 5    20.01.2023    08:39
Не работают
1. Получить факультет студента
2. Получить студентов факультета

Commit 4    19.01.2023    23:39
Устранил некорректный импорт

Commit 3    18.01.2023    23:35
Подключена база данных.

Commit 2    17.01.2023    18:51

leushkinml committed:
Commit 12    03.02.2023    14:20
SQL запросы

Commit 11    02.02.2023    15:12

Commit 10    30.01.2023    20:39

Commit 9    28.01.2023    22:43
Тесты контроллеров

Commit 8    26.01.2023    18:10
Потоки данных. Работа с файлами.

Commit 7    23.01.2023    20:07
ДЗ выполнено.

Commit 6    22.01.2023    20:28
ДЗ выполнено.

Commit 5    20.01.2023    08:39
Не работают
1. Получить факультет студента
2. Получить студентов факультета

Commit 4    19.01.2023    23:39
Устранил некорректный импорт

Commit 3    18.01.2023    23:35
Подключена база данных.

Commit 2    17.01.2023    18:51

leushkinml committed:
Commit 12    03.02.2023    14:20
SQL запросы

Commit 11    02.02.2023    15:12

Commit 10    30.01.2023    20:39

Commit 9    28.01.2023    22:43
Тесты контроллеров

Commit 8    26.01.2023    18:10
Потоки данных. Работа с файлами.

Commit 7    23.01.2023    20:07
ДЗ выполнено.

Commit 6    22.01.2023    20:28
ДЗ выполнено.

Commit 5    20.01.2023    08:39
Не работают
1. Получить факультет студента
2. Получить студентов факультета

Commit 4    19.01.2023    23:39
Устранил некорректный импорт

Commit 3    18.01.2023    23:35
Подключена база данных.

Commit 2    17.01.2023    18:51

leushkinml committed:
Commit 12    03.02.2023    14:20
SQL запросы

Commit 11    02.02.2023    15:12

Commit 10    30.01.2023    20:39

Commit 9    28.01.2023    22:43
Тесты контроллеров

Commit 8    26.01.2023    18:10
Потоки данных. Работа с файлами.

Commit 7    23.01.2023    20:07
ДЗ выполнено.

Commit 6    22.01.2023    20:28
ДЗ выполнено.

Commit 5    20.01.2023    08:39
Не работают
1. Получить факультет студента
2. Получить студентов факультета

Commit 4    19.01.2023    23:39
Устранил некорректный импорт

Commit 3    18.01.2023    23:35
Подключена база данных.

Commit 2    17.01.2023    18:51

leushkinml committed:
Commit 12    03.02.2023    14:20
SQL запросы

Commit 11    02.02.2023    15:12

Commit 10    30.01.2023    20:39

Commit 9    28.01.2023    22:43
Тесты контроллеров

Commit 8    26.01.2023    18:10
Потоки данных. Работа с файлами.

Commit 7    23.01.2023    20:07
ДЗ выполнено.

Commit 6    22.01.2023    20:28
ДЗ выполнено.

Commit 5    20.01.2023    08:39
Не работают
1. Получить факультет студента
2. Получить студентов факультета

Commit 4    19.01.2023    23:39
Устранил некорректный импорт

Commit 3    18.01.2023    23:35
Подключена база данных.

Commit 2    17.01.2023    18:51

leushkinml committed:
Commit 13    06.02.2023    18:40
Управление схемами баз данных

Commit 12    03.02.2023    14:20
SQL запросы

Commit 11    02.02.2023    15:12

Commit 10    30.01.2023    20:39

Commit 9    28.01.2023    22:43
Тесты контроллеров

Commit 8    26.01.2023    18:10
Потоки данных. Работа с файлами.

Commit 7    23.01.2023    20:07
ДЗ выполнено.

Commit 6    22.01.2023    20:28
ДЗ выполнено.

Commit 5    20.01.2023    08:39
Не работают
1. Получить факультет студента
2. Получить студентов факультета

Commit 4    19.01.2023    23:39
Устранил некорректный импорт

Commit 3    18.01.2023    23:35
Подключена база данных.

Commit 2    17.01.2023    18:51

leushkinml committed:
Commit 13    06.02.2023    18:40
Управление схемами баз данных

Commit 12    03.02.2023    14:20
SQL запросы

Commit 11    02.02.2023    15:12

Commit 10    30.01.2023    20:39

Commit 9    28.01.2023    22:43
Тесты контроллеров

Commit 8    26.01.2023    18:10
Потоки данных. Работа с файлами.

Commit 7    23.01.2023    20:07
ДЗ выполнено.

Commit 6    22.01.2023    20:28
ДЗ выполнено.

Commit 5    20.01.2023    08:39
Не работают
1. Получить факультет студента
2. Получить студентов факультета

Commit 4    19.01.2023    23:39
Устранил некорректный импорт

Commit 3    18.01.2023    23:35
Подключена база данных.

Commit 2    17.01.2023    18:51

leushkinml committed:
Merge remote-tracking branch 'origin/HW1' into HW1

# Conflicts:
#	src/main/java/ru/hogwarts/schoollion/controller/AvatarController.java
#	src/main/java/ru/hogwarts/schoollion/controller/StudentController.java
#	src/test/java/ru/hogwarts/schoollion/controller/StudentControllerTest.java

leushkinml committed:
Commit 14    06.02.2023    23:19
Миграции и Индексы

Commit 13    06.02.2023    18:40
Управление схемами баз данных

Commit 12    03.02.2023    14:20
SQL запросы

Commit 11    02.02.2023    15:12

Commit 10    30.01.2023    20:39

Commit 9    28.01.2023    22:43
Тесты контроллеров

Commit 8    26.01.2023    18:10
Потоки данных. Работа с файлами.

Commit 7    23.01.2023    20:07
ДЗ выполнено.

Commit 6    22.01.2023    20:28
ДЗ выполнено.

Commit 5    20.01.2023    08:39
Не работают
1. Получить факультет студента
2. Получить студентов факультета

Commit 4    19.01.2023    23:39
Устранил некорректный импорт

Commit 3    18.01.2023    23:35
Подключена база данных.

Commit 2    17.01.2023    18:51

leushkinml committed:
Commit 14    06.02.2023    23:19
Миграции и Индексы

Commit 13    06.02.2023    18:40
Управление схемами баз данных

Commit 12    03.02.2023    14:20
SQL запросы

Commit 11    02.02.2023    15:12

Commit 10    30.01.2023    20:39

Commit 9    28.01.2023    22:43
Тесты контроллеров

Commit 8    26.01.2023    18:10
Потоки данных. Работа с файлами.

Commit 7    23.01.2023    20:07
ДЗ выполнено.

Commit 6    22.01.2023    20:28
ДЗ выполнено.

Commit 5    20.01.2023    08:39
Не работают
1. Получить факультет студента
2. Получить студентов факультета

Commit 4    19.01.2023    23:39
Устранил некорректный импорт

Commit 3    18.01.2023    23:35
Подключена база данных.

Commit 2    17.01.2023    18:51

leushkinml committed:
Commit 14    06.02.2023    23:19
Миграции и Индексы

Commit 13    06.02.2023    18:40
Управление схемами баз данных

Commit 12    03.02.2023    14:20
SQL запросы

Commit 11    02.02.2023    15:12

Commit 10    30.01.2023    20:39

Commit 9    28.01.2023    22:43
Тесты контроллеров

Commit 8    26.01.2023    18:10
Потоки данных. Работа с файлами.

Commit 7    23.01.2023    20:07
ДЗ выполнено.

Commit 6    22.01.2023    20:28
ДЗ выполнено.

Commit 5    20.01.2023    08:39
Не работают
1. Получить факультет студента
2. Получить студентов факультета

Commit 4    19.01.2023    23:39
Устранил некорректный импорт

Commit 3    18.01.2023    23:35
Подключена база данных.

Commit 2    17.01.2023    18:51

leushkinml committed:
Commit 14    06.02.2023    23:19
Миграции и Индексы

Commit 13    06.02.2023    18:40
Управление схемами баз данных

Commit 12    03.02.2023    14:20
SQL запросы

Commit 11    02.02.2023    15:12

Commit 10    30.01.2023    20:39

Commit 9    28.01.2023    22:43
Тесты контроллеров

Commit 8    26.01.2023    18:10
Потоки данных. Работа с файлами.

Commit 7    23.01.2023    20:07
ДЗ выполнено.

Commit 6    22.01.2023    20:28
ДЗ выполнено.

Commit 5    20.01.2023    08:39
Не работают
1. Получить факультет студента
2. Получить студентов факультета

Commit 4    19.01.2023    23:39
Устранил некорректный импорт

Commit 3    18.01.2023    23:35
Подключена база данных.

Commit 2    17.01.2023    18:51

leushkinml committed:
Commit 15    09.02.2023    18:54

Commit 14    06.02.2023    23:19
Миграции и Индексы

Commit 13    06.02.2023    18:40
Управление схемами баз данных

Commit 12    03.02.2023    14:20
SQL запросы

Commit 11    02.02.2023    15:12

Commit 10    30.01.2023    20:39

Commit 9    28.01.2023    22:43
Тесты контроллеров

Commit 8    26.01.2023    18:10
Потоки данных. Работа с файлами.

Commit 7    23.01.2023    20:07
ДЗ выполнено.

Commit 6    22.01.2023    20:28
ДЗ выполнено.

Commit 5    20.01.2023    08:39
Не работают
1. Получить факультет студента
2. Получить студентов факультета

Commit 4    19.01.2023    23:39
Устранил некорректный импорт

Commit 3    18.01.2023    23:35
Подключена база данных.

Commit 2    17.01.2023    18:51

leushkinml committed:
Commit 15    09.02.2023    18:54
Логгирование и конфигурирование

Commit 14    06.02.2023    23:19
Миграции и Индексы

Commit 13    06.02.2023    18:40
Управление схемами баз данных

Commit 12    03.02.2023    14:20
SQL запросы

Commit 11    02.02.2023    15:12

Commit 10    30.01.2023    20:39

Commit 9    28.01.2023    22:43
Тесты контроллеров

Commit 8    26.01.2023    18:10
Потоки данных. Работа с файлами.

Commit 7    23.01.2023    20:07
ДЗ выполнено.

Commit 6    22.01.2023    20:28
ДЗ выполнено.

Commit 5    20.01.2023    08:39
Не работают
1. Получить факультет студента
2. Получить студентов факультета

Commit 4    19.01.2023    23:39
Устранил некорректный импорт

Commit 3    18.01.2023    23:35
Подключена база данных.

Commit 2    17.01.2023    18:51

leushkinml committed:
Commit 15    09.02.2023    18:54
Логгирование и конфигурирование

Commit 14    06.02.2023    23:19
Миграции и Индексы

Commit 13    06.02.2023    18:40
Управление схемами баз данных

Commit 12    03.02.2023    14:20
SQL запросы

Commit 11    02.02.2023    15:12

Commit 10    30.01.2023    20:39

Commit 9    28.01.2023    22:43
Тесты контроллеров

Commit 8    26.01.2023    18:10
Потоки данных. Работа с файлами.

Commit 7    23.01.2023    20:07
ДЗ выполнено.

Commit 6    22.01.2023    20:28
ДЗ выполнено.

Commit 5    20.01.2023    08:39
Не работают
1. Получить факультет студента
2. Получить студентов факультета

Commit 4    19.01.2023    23:39
Устранил некорректный импорт

Commit 3    18.01.2023    23:35
Подключена база данных.

Commit 2    17.01.2023    18:51

leushkinml committed:
Commit 15    09.02.2023    18:54
Логгирование и конфигурирование

Commit 14    06.02.2023    23:19
Миграции и Индексы

Commit 13    06.02.2023    18:40
Управление схемами баз данных

Commit 12    03.02.2023    14:20
SQL запросы

Commit 11    02.02.2023    15:12

Commit 10    30.01.2023    20:39

Commit 9    28.01.2023    22:43
Тесты контроллеров

Commit 8    26.01.2023    18:10
Потоки данных. Работа с файлами.

Commit 7    23.01.2023    20:07
ДЗ выполнено.

Commit 6    22.01.2023    20:28
ДЗ выполнено.

Commit 5    20.01.2023    08:39
Не работают
1. Получить факультет студента
2. Получить студентов факультета

Commit 4    19.01.2023    23:39
Устранил некорректный импорт

Commit 3    18.01.2023    23:35
Подключена база данных.

Commit 2    17.01.2023    18:51

leushkinml committed:
Commit 15    09.02.2023    18:54
Логгирование и конфигурирование

Commit 14    06.02.2023    23:19
Миграции и Индексы

Commit 13    06.02.2023    18:40
Управление схемами баз данных

Commit 12    03.02.2023    14:20
SQL запросы

Commit 11    02.02.2023    15:12

Commit 10    30.01.2023    20:39

Commit 9    28.01.2023    22:43
Тесты контроллеров

Commit 8    26.01.2023    18:10
Потоки данных. Работа с файлами.

Commit 7    23.01.2023    20:07
ДЗ выполнено.

Commit 6    22.01.2023    20:28
ДЗ выполнено.

Commit 5    20.01.2023    08:39
Не работают
1. Получить факультет студента
2. Получить студентов факультета

Commit 4    19.01.2023    23:39
Устранил некорректный импорт

Commit 3    18.01.2023    23:35
Подключена база данных.

Commit 2    17.01.2023    18:51

leushkinml committed:
Commit 16    17.02.2023    22:31
Работа со Stream API

Commit 15    09.02.2023    18:54
Логгирование и конфигурирование

Commit 14    06.02.2023    23:19
Миграции и Индексы

Commit 13    06.02.2023    18:40
Управление схемами баз данных

Commit 12    03.02.2023    14:20
SQL запросы

Commit 11    02.02.2023    15:12

Commit 10    30.01.2023    20:39

Commit 9    28.01.2023    22:43
Тесты контроллеров

Commit 8    26.01.2023    18:10
Потоки данных. Работа с файлами.

Commit 7    23.01.2023    20:07
ДЗ выполнено.

Commit 6    22.01.2023    20:28
ДЗ выполнено.

Commit 5    20.01.2023    08:39
Не работают
1. Получить факультет студента
2. Получить студентов факультета

Commit 4    19.01.2023    23:39
Устранил некорректный импорт

Commit 3    18.01.2023    23:35
Подключена база данных.

Commit 2    17.01.2023    18:51

leushkinml committed:
Commit 16    17.02.2023    22:31
Работа со Stream API

Commit 15    09.02.2023    18:54
Логгирование и конфигурирование

Commit 14    06.02.2023    23:19
Миграции и Индексы

Commit 13    06.02.2023    18:40
Управление схемами баз данных

Commit 12    03.02.2023    14:20
SQL запросы

Commit 11    02.02.2023    15:12

Commit 10    30.01.2023    20:39

Commit 9    28.01.2023    22:43
Тесты контроллеров

Commit 8    26.01.2023    18:10
Потоки данных. Работа с файлами.

Commit 7    23.01.2023    20:07
ДЗ выполнено.

Commit 6    22.01.2023    20:28
ДЗ выполнено.

Commit 5    20.01.2023    08:39
Не работают
1. Получить факультет студента
2. Получить студентов факультета

Commit 4    19.01.2023    23:39
Устранил некорректный импорт

Commit 3    18.01.2023    23:35
Подключена база данных.

Commit 2    17.01.2023    18:51

dmitry-bizin commented on src/main/java/ru/hogwarts/schoollion/controller/StudentController.java:
```diff
@@ -0,0 +1,182 @@
+
+    @GetMapping("/students-with-name-start-A")
+    public ResponseEntity<List<String>> getStudentsWithNameStartWithA() {
+        List<String> students = studentService.getStudentsWithNameStartWithA();
```
здесь просто можно вызвать метод `getStudentsWithNameStartWith` и передать в него "A", чтобы не дублировать код.

dmitry-bizin commented on src/main/java/ru/hogwarts/schoollion/service/StudentService.java:
```diff
@@ -0,0 +1,127 @@
+        return studentsSorted;
+    }
+
+    public List<String> getStudentsWithNameStartWithA() {
```
дублирование кода, метод можно удалить и использовать только getStudentsWithNameStartWith

dmitry-bizin commented on src/main/java/ru/hogwarts/schoollion/service/StudentService.java:
```diff
@@ -0,0 +1,127 @@
+    public Double studentsAverageAge() {
+        List<Student> students = new ArrayList<>(studentRepository.findAll());
+        Double studentsAverageAge = students.stream().
+                mapToInt(Student::getAge).average().getAsDouble();
```
неправильная работа с OptionalDouble, если студентов не будет, то здесь возникнет исключение, вместо getAsDouble можно использовать `orElse(0D)`

dmitry-bizin commented on src/main/java/ru/hogwarts/schoollion/service/FacultyService.java:
```diff
@@ -0,0 +1,147 @@
+        String longestFacultyName = users.stream()
+                .map(u -> u.getName())
+                .sorted(Comparator.comparing(String::length).reversed())
+                .findFirst().orElse("1");
```
`orElse(null)` лучше

dmitry-bizin commented on src/main/java/ru/hogwarts/schoollion/controller/FacultyController.java:
```diff
@@ -0,0 +1,165 @@
+    @GetMapping("/faculty-with-longest-name")
+    public ResponseEntity<Optional<String>> getFacultyWithLongestName() {
+        Optional<String> faculty = facultyService.getFacultyWithLongestName();
+        return ResponseEntity.ok(faculty);
```
здесь можно использовать метод of вместо ok

dmitry-bizin commented on src/main/java/ru/hogwarts/schoollion/controller/FacultyController.java:
```diff
@@ -0,0 +1,165 @@
+
+    // Работа со Stream API
+    @GetMapping("/faculty-with-longest-name")
+    public ResponseEntity<Optional<String>> getFacultyWithLongestName() {
```
а здесь просто `ResponseEntity<String>`

dmitry-bizin commented on src/main/java/ru/hogwarts/schoollion/service/FacultyService.java:
```diff
@@ -0,0 +1,147 @@
+        long finish3 = System.nanoTime();
+        long elapsed3 = finish3 - start3;
+
+        return "1-ый метод: Время вычисления,нс: " + elapsed1 + ". Результат вычисления: " + sum1 +
```
ну и что, помог parallel-то?)

dmitry-bizin replied:
не думаю

dmitry-bizin replied:
рекомендую почитать вот [эту](https://www.baeldung.com/java-when-to-use-parallel-stream) статью для объяснения

leushkinml committed:
Commit 16    17.02.2023    22:31
Работа со Stream API

Commit 15    09.02.2023    18:54
Логгирование и конфигурирование

Commit 14    06.02.2023    23:19
Миграции и Индексы

Commit 13    06.02.2023    18:40
Управление схемами баз данных

Commit 12    03.02.2023    14:20
SQL запросы

Commit 11    02.02.2023    15:12

Commit 10    30.01.2023    20:39

Commit 9    28.01.2023    22:43
Тесты контроллеров

Commit 8    26.01.2023    18:10
Потоки данных. Работа с файлами.

Commit 7    23.01.2023    20:07
ДЗ выполнено.

Commit 6    22.01.2023    20:28
ДЗ выполнено.

Commit 5    20.01.2023    08:39
Не работают
1. Получить факультет студента
2. Получить студентов факультета

Commit 4    19.01.2023    23:39
Устранил некорректный импорт

Commit 3    18.01.2023    23:35
Подключена база данных.

Commit 2    17.01.2023    18:51

leushkinml committed:
Commit 16    20.02.2023    01:08
Работа с Потоком и Синхронизацией

Commit 16    17.02.2023    22:31
Работа со Stream API

Commit 15    09.02.2023    18:54
Логгирование и конфигурирование

Commit 14    06.02.2023    23:19
Миграции и Индексы

Commit 13    06.02.2023    18:40
Управление схемами баз данных

Commit 12    03.02.2023    14:20
SQL запросы

Commit 11    02.02.2023    15:12

Commit 10    30.01.2023    20:39

Commit 9    28.01.2023    22:43
Тесты контроллеров

Commit 8    26.01.2023    18:10
Потоки данных. Работа с файлами.

Commit 7    23.01.2023    20:07
ДЗ выполнено.

Commit 6    22.01.2023    20:28
ДЗ выполнено.

Commit 5    20.01.2023    08:39
Не работают
1. Получить факультет студента
2. Получить студентов факультета

Commit 4    19.01.2023    23:39
Устранил некорректный импорт

Commit 3    18.01.2023    23:35
Подключена база данных.

Commit 2    17.01.2023    18:51

leushkinml committed:
Commit 16    20.02.2023    01:08
Работа с Потоком и Синхронизацией

Commit 16    17.02.2023    22:31
Работа со Stream API

Commit 15    09.02.2023    18:54
Логгирование и конфигурирование

Commit 14    06.02.2023    23:19
Миграции и Индексы

Commit 13    06.02.2023    18:40
Управление схемами баз данных

Commit 12    03.02.2023    14:20
SQL запросы

Commit 11    02.02.2023    15:12

Commit 10    30.01.2023    20:39

Commit 9    28.01.2023    22:43
Тесты контроллеров

Commit 8    26.01.2023    18:10
Потоки данных. Работа с файлами.

Commit 7    23.01.2023    20:07
ДЗ выполнено.

Commit 6    22.01.2023    20:28
ДЗ выполнено.

Commit 5    20.01.2023    08:39
Не работают
1. Получить факультет студента
2. Получить студентов факультета

Commit 4    19.01.2023    23:39
Устранил некорректный импорт

Commit 3    18.01.2023    23:35
Подключена база данных.

Commit 2    17.01.2023    18:51
